stm32-lvgl/README.md

```diff
@@ -0,0 +1,50 @@
+# stm32-lvgl
+
+<img src="assets/video.mov" />
+
+This sample code is demonstrating a "full" graphical firmware running on an STM32 microcontroller board, concretely the STM32F746G "Discovery" board. It shows using:
+
+- **ELF file format**, linking with lld, with a custom simple linker script (and thus it builds identically on both macOS and Linux hosts)
+- **LLVM Embedded Toolchain for ARM**
+- **LVGL** graphical/input/animation library
+- The **DRAM, LCD, touch panel, GPIO pins and interrupts** on the STM32F746G
+- **No other SDKs or library dependencies** -- all the startup code, including MCU, board and peripheral initialization is done in Swift source code
+
+Additionally, this sample code:
+
+- Has **LSP integration** set up via the `.sourcekit-lsp/config.json` file, confirmed to work in multiple code editors (VS Code, Sublime Text, Zed)
+- Uses **SwiftPM's toolset.json** to define compiler and linker flags
+- Has a host OS (macOS, Linux) **"simulator" using SDL** that can use the same "business logic" code to render the same LVGL UI.
+
+## How to build and run this example:
+
+1. Connect the STM32F746G-DISCO board via the ST-LINK USB port to your Mac.
+2. Install a recent nightly "main" Swift toolchain. This is best done using `swiftly`, for example with:
+```console
+$ swiftly install main-snapshot-2025-04-12
```

Comment:
I think we should tell people to install swiftly and instead rely on the .swift-version in this repo. This allows users to download the repo and not have to worry about manually selecting a compatible toolchain. Thats our/ci's responsibility.

Reply:
updated the README